controllers/argocd/status.go

```diff
@@ -222,6 +226,30 @@ func (r *ReconcileArgoCD) reconcileStatusServer(cr *argoprojv1a1.ArgoCD) error {
 	return nil
 }
 
+// reconcileStatusNotifications will ensure that the Notifications status is updated for the given ArgoCD.
+func (r *ReconcileArgoCD) reconcileStatusNotifications(cr *argoprojv1a1.ArgoCD) error {
+	status := "Unknown"
+
+	deploy := newDeploymentWithSuffix("notifications-controller", "controller", cr)
+	if argoutil.IsObjectFound(r.Client, cr.Namespace, deploy.Name, deploy) {
+		status = "Running"
```

Comment:
When the deployment is found, the other status reconciler methods only set the status to "Running" when the `deploy.Status.ReadyReplicas` is the same as `deploy.Spec.Replicas`, otherwise they set it to "Pending" -- is there a reason not to do the same for the status of the notificationsController?

Reply:
@jopit I wasn't quite sure if there was a use case for us to support multiple replicas of the notifications-controller at the time
However I realize now these are 2 different checks, so I will update the logic here (which should default to 1 replica within the deployment, unless we decide to allow users to be able to express the replica count for the notifications-controller through the CR at some point)
Thanks for catching this!